pkg/cloudprovider/azure/volume_snapshotter.go

```diff
@@ -185,6 +203,58 @@ func (b *VolumeSnapshotter) CreateVolumeFromSnapshot(snapshotID, volumeType, vol
 	return diskName, nil
 }
 
+func (b *VolumeSnapshotter) snapshotWhenAvailable(snapshotID string) (disk.Snapshot, error) {
+	logger := b.log.WithField("snapshotID", snapshotID)
+
+	var snapshot disk.Snapshot
+	snapshotIdentifier, err := parseFullSnapshotName(snapshotID)
+	if err != nil {
+		return snapshot, err
+	}
+	err = wait.PollImmediate(time.Second, b.snapshotCreationTimeout, func() (bool, error) {
+		var err error
+		snapshot, err = b.snaps.Get(context.TODO(), snapshotIdentifier.resourceGroup, snapshotIdentifier.name)
+		if err != nil {
+			return true, err
+		}
+		// FIXME: ProvisioningState is a named type with const status values as of version "2019-03-01"
+		// which is not available in the current "2018-04-01" version. Use these consts on API dep upgrade
+		if snapshot.DiskProperties == nil {
+			logger.Debug("snapshot has nil DiskProperties")
+			return true, errors.Errorf("Snapshot has nil DiskProperties")
+		}
+		if snapshot.DiskProperties.ProvisioningState == nil {
+			logger.Debug("snapshot has nil ProvisioningState")
+			return true, errors.Errorf("Snapshot has nil ProvisioningState")
+		}
+		if *snapshot.DiskProperties.ProvisioningState == "Creating" ||
+			*snapshot.DiskProperties.ProvisioningState == "Migrating" ||
+			*snapshot.DiskProperties.ProvisioningState == "Updating" {
+			logger.Debug("snapshot not yet ready for use")
+			return false, nil
+		}
+		if *snapshot.DiskProperties.ProvisioningState == "Succeeded" {
+			return true, nil
+		}
+		if *snapshot.DiskProperties.ProvisioningState == "Failed" {
+			logger.Debug("snapshot has 'Failed' ProvisioningState")
+			return true, errors.Errorf("Snapshot has 'Failed' ProvisioningState")
+		}
+		if *snapshot.DiskProperties.ProvisioningState == "Failed" {
```

Comment:
@sseago Is this intended? should it be Deleting?

Reply:
@pranavgaikwad fixed